### ISSUE TITLE
Suggest inserting using declarations in addition to inserting qualifiers for unknown/unresolved symbols

### ISSUE THREAD

**Issue by kadircet** [clangd/clangd#976]
Clangd already suggests qualifying symbols e.g:
```
namespace ns { struct Foo {}; }
void bar() {
  Foo f; // here clangd will offer a fix for qualifting `Foo` to `ns::Foo`.
}
```

In addition to that, we can also offer a fix that inserts a using declaration.

**Comment by sam-mccall** [comment]
Yes, I want this *all the time* and always do the dumb two-step version.

The trick is getting the layering right: FindUsing doesn't know about the namespace (though I guess the recovered AST has it...) and Sema doesn't know where to insert using.
